src/test/java/org/openrewrite/java/testing/junit6/MinimumJreConditionsTest.java

```diff
@@ -0,0 +1,843 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.junit6;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class MinimumJreConditionsTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec
+          .parser(JavaParser.fromJavaVersion()
+            .classpathFromResources(new InMemoryExecutionContext(), "junit-jupiter-api"))
+          .recipe(new MinimumJreConditions("17"));
+    }
+
+    @Nested
+    class EnabledOnJreTests {
+        @ParameterizedTest
+        @ValueSource(strings = {
+          "JAVA_11",
+          "JAVA_8",
+          "JRE.JAVA_11",
+          "JRE.JAVA_8",
+          "value = JRE.JAVA_11",
+          "value = JRE.JAVA_8",
+          "versions = { 11 }",
+          "versions = { 8 }"
+        })
+        void removeTestEnabledOnOlderJre(String jre) {
+            rewriteRun(
+              java(
+                """
+                  import org.junit.jupiter.api.Test;
+                  import org.junit.jupiter.api.condition.EnabledOnJre;
+                  import org.junit.jupiter.api.condition.JRE;
+
+                  import static org.junit.jupiter.api.condition.JRE.*;
+
+                  class MyTest {
+                      @Test
+                      @EnabledOnJre(%s)
+                      void testOnlyOnJava8() {
+                          System.out.println("Java 8");
+                      }
+
+                      @Test
+                      void normalTest() {
+                          System.out.println("Any version");
+                      }
+                  }
+                  """.formatted(jre),
+                """
+                  import org.junit.jupiter.api.Test;
+                  import org.junit.jupiter.api.condition.EnabledOnJre;
+                  import org.junit.jupiter.api.condition.JRE;
+
+                  import static org.junit.jupiter.api.condition.JRE.*;
```

Comment:
I'd have hoped to see these now unused imports removed, as opposed to having those linger after a removal.